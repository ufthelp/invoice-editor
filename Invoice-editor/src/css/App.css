.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

/*Invoice Component CSS */
.Invoice {
    max-width: 50%;
    margin: 0 auto;
    border: 1px solid #ccc;
    background: #fff;
    padding: 25px 0;
}
.Invoice-header {
    display: flex;
    justify-content: space-around;
    padding: 20px 0;
    border-bottom: 1px solid #ccc;
}
