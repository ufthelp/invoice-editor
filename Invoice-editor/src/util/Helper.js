

module.exports ={
    isValidNumber: function(val){
    if(val !=='' && val !== undefined && val !== null && val !==0){
            return true;
        }
        return false;
    },
    
    getComputedValues: function(invoiceList, index){
        let subTotalAmt = 0;
        let length = invoiceList.length
        for(let i = 0 ; i < length ; i++){
            subTotalAmt += Number(invoiceList[i].total);
        }
        
        let taxAmt = parseFloat(.05 * subTotalAmt).toFixed(2);
        let grandTotalAmt =parseFloat((+taxAmt) + (+subTotalAmt)).toFixed(2);
        
        return{
            subTotal: subTotalAmt,
            taxAmount: taxAmt,
            grandTotal: grandTotalAmt
        }
    }

};