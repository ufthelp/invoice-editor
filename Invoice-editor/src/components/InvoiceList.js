import React, { Component } from 'react';
import InvoiceRow from './InvoiceRow'
import '../css/invoiceList.css'

/*Invoice list component creates Invoice Headers
***Called Components :InvoiceRow 
*/

class InvoiceList extends Component{
    render(){
        return(
            <div className="invoice">
                <section className="invoice-header">
                    <div>Item</div>
                    <div>Qty</div>
                    <div>Price </div>
                    <div>Total</div>
                </section>
                <InvoiceRow/>  
            </div> 


        )
    }
}


export default InvoiceList;