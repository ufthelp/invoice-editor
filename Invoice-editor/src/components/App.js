import React, { Component } from 'react';
import InvoiceList from './InvoiceList'
import logo from '../assets/logo.svg';
import '../css/App.css';

/*Main App Level Component
***Called Components :InvoiceList 
*/

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to Invoice Editor</h2>
        </div>
        <InvoiceList/>  
      </div>
    );
  }
}


export default App;
