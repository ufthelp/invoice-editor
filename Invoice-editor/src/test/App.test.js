import React from 'react';
import ReactDOM from 'react-dom';
import Renderer from 'react-test-renderer'
import App from '../components/App';
import InvoiceList from '../components/InvoiceList';
import util from '../util/Helper'

/*App Level Test*/
describe('Test the App rendering and Snapshot',() =>{
	it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<App />, div);
	});

	test('Test App Snapshot',() =>{
		const component = Renderer.create(<App/>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();

	});
});
/*Test cases for Helper methods*/
describe('Helper Method - isValidNumber',() =>{
	test('number is valid' ,() =>{
		expect(util.isValidNumber(2)).toBeTruthy();
	});

	test('number is null' ,() =>{
		expect(util.isValidNumber(null)).toBeFalsy();
	});
});

describe('Helper Method - getComputedValues - Returns Subtotal, taxAmount & Grand Total of items',() =>{
	let arrItems =[ {
			qty : 2,
			price: 1000,
			total : 2000
		},
		{
			qty : 3,
			price: 3000,
			total : 9000
		}
	];

	test('Values are returned' ,() =>{
		expect(util.getComputedValues(arrItems,0)).toBeDefined();
		
	});

	test('Printing grand total' ,() =>{
		let res = util.getComputedValues(arrItems,0)
		expect(res.grandTotal).toEqual('11550.00');
			
	});

	test('Printing Sub Total' ,() =>{
		let res = util.getComputedValues(arrItems,0)
		expect(res.subTotal).toEqual(11000);
	});

	test('Printing Tax Amount' ,() =>{
		let res = util.getComputedValues(arrItems,0)
		expect(res.taxAmount).toEqual('550.00');
	});



});

/*Test cases for Invoice List*/
it('Invoice List component is generated', () => {
	const div = document.createElement('div');
	ReactDOM.render(<InvoiceList />, div);
});


