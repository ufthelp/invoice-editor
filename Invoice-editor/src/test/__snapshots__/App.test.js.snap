// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the App rendering and Snapshot Test App Snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to Invoice Editor
    </h2>
  </div>
  <div
    className="invoice"
  >
    <section
      className="invoice-header"
    >
      <div>
        Item
      </div>
      <div>
        Qty
      </div>
      <div>
        Price 
      </div>
      <div>
        Total
      </div>
    </section>
    <form>
      <div
        className="item-row"
      >
        <input
          className="form-item"
          defaultValue=""
          id="item"
          placeholder="Description of Item..."
          required={true}
          type="text"
        />
        <input
          className="form-item form-qty"
          name="qty"
          onChange={[Function]}
          required={true}
          type="number"
          value={0}
        />
        <input
          className="form-item"
          name="price"
          onChange={[Function]}
          required={true}
          type="number"
          value={0}
        />
        <input
          className="form-item"
          name="total"
          required={true}
          type="text"
          value={0}
        />
        <button
          className="delete-btn "
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <button
        className="add-btn"
        onClick={[Function]}
        type="button"
      >
        Add Item
      </button>
      <section
        className="invoice-total"
      >
        <section
          className="invoice-p-labels"
        >
          <div
            className="item-label"
          >
            Subtotal
          </div>
          <div
            className="item-label"
          >
            Tax (5%)
          </div>
          <div
            className="item-label"
          >
            Total
          </div>
        </section>
        <section
          className="invoice-cal"
        >
          <label
            value={true}
          >
            $
          </label>
          <label
            value={true}
          >
            $
          </label>
          <label
            value={true}
          >
            $
          </label>
        </section>
      </section>
    </form>
  </div>
</div>
`;
